node/tools/src/k8s.rs

```diff
@@ -39,61 +44,53 @@ pub async fn get_consensus_nodes_address(client: &Client) -> anyhow::Result<Vec<
         !pods.items.is_empty(),
         "No consensus pods found in the k8s cluster"
     );
-    let pod_addresses: Vec<SocketAddr> = pods
-        .into_iter()
-        .filter_map(|pod| {
-            let pod_spec = pod.spec.context("Failed to get pod spec").ok()?;
-            let pod_running_container = pod_spec
-                .containers
-                .first()
-                .context("Failed to get pod container")
-                .ok()?
-                .to_owned();
-            let docker_image = pod_running_container
-                .image
-                .context("Failed to get pod docker image")
-                .ok()?;
-
-            if docker_image.contains(DOCKER_IMAGE_NAME) {
-                let pod_ip = pod
-                    .status
-                    .context("Failed to get pod status")
-                    .ok()?
-                    .pod_ip
-                    .context("Failed to get pod ip")
-                    .ok()?;
-                let port = pod_running_container.ports?.iter().find_map(|port| {
-                    let port = port.container_port.try_into().ok()?;
+    let mut node_rpc_addresses: Vec<SocketAddr> = Vec::new();
+    for pod in pods.into_iter() {
+        let pod_spec = pod.spec.as_ref().context("Failed to get pod spec")?;
+        let pod_container = pod_spec
+            .containers
+            .first()
+            .context("Failed to get container")?;
+        if pod_container
+            .image
+            .as_ref()
+            .context("Failed to get image")?
+            .contains(DOCKER_IMAGE_NAME)
+        {
+            let pod_ip = pod
+                .status
+                .context("Failed to get pod status")?
+                .pod_ip
+                .context("Failed to get pod ip")?;
+            let pod_rpc_port = pod_container
+                .ports
+                .as_ref()
+                .context("Failed to get ports of container")?
+                .iter()
+                .find_map(|port| {
+                    let port: u16 = port.container_port.try_into().ok()?;
```

Comment:
shouln't failed conversion return an error?

Reply:
The overall operation will fail due to the `.context("Failed parsing container port")?;` below. If the conversion fails then a `None` is returned, if every port returns `None` in the `find_map` operation, the context will raise an error because there was no port that met the criteria. While it's true that if the conversion succeeds but there's no port different from `NODE_PORT`, the error saying `Failed parsing container port` will also be thrown. I can separate the `find` and `map` operations, and that should fix it, but let me know what you think.

Reply:
wrt correctness, this should fail if anything in the response received is different than expected, even if the given part of the response is unused. Said that, it is not that important in testing code, so up to you.